tests/shutdown/cleanup_before_shutdown.pm

```diff
@@ -21,6 +21,14 @@ use version_utils;
 
 sub run {
     select_console('root-console');
+    # Collect detailed logs to investigate shutdown issues and redirect them to serial console.
+    if (get_var('DEBUG_SHUTDOWN')) {
+        my $shutdown_debug_file         = '/usr/lib/systemd/system-shutdown/debug.sh';
+        my $shutdown_debug_file_content = "#!/bin/sh
+dmesg > /dev/$serialdev";
+        type_string("cat > $shutdown_debug_file <<EOF\n$shutdown_debug_file_content\nEOF\n");
```

Comment:
Why not just

```perl
assert_script_run "echo \"dmesg > /dev/$serialdev\" > /usr/lib/systemd/system-shutdown/debug.sh";
```

instead of the above three lines

Reply:
Your line doesn't work, but I've changed the code to one line to be:
```perl
assert_script_run "echo -e '#!/bin/sh\\ndmesg > /dev/$serialdev' > /usr/lib/systemd/system-shutdown/debug.sh";
```